micro_sam/__init__.py

```diff
@@ -5,4 +5,22 @@
 .. include:: ../doc/python_library.md
 """
 
-from .__version__ import __version__
+__version__ = "0.1.2.post1"
```

Comment:
I don't know why, but if I kept the line `from .__version__ import __version__` here then it would not install with pip. It seemed to be trying to import the whole micro-sam package instead of just the version string we wanted.

If you know how to avoid this problem, please let me know.

<details><summary>Details</summary>
```
$ pip install -e .

Obtaining file:///Users/genevieb/Documents/GitHub/micro-sam
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Preparing editable metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [59 lines of output]
      Traceback (most recent call last):
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/config/expand.py", line 81, in __getattr__
          return next(
      StopIteration

      The above exception was the direct cause of the following exception:

      Traceback (most recent call last):
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/config/expand.py", line 193, in read_attr
          return getattr(StaticModule(module_name, spec), attr_name)
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/config/expand.py", line 87, in __getattr__
          raise AttributeError(f"{self.name} has no attribute {attr}") from e
      AttributeError: micro_sam has no attribute __version__

      During handling of the above exception, another exception occurred:

      Traceback (most recent call last):
        File "/Users/genevieb/mambaforge/envs/napari/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/genevieb/mambaforge/envs/napari/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/Users/genevieb/mambaforge/envs/napari/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 181, in prepare_metadata_for_build_editable
          return hook(metadata_directory, config_settings)
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 473, in prepare_metadata_for_build_editable
          return self.prepare_metadata_for_build_wheel(
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 396, in prepare_metadata_for_build_wheel
          self.run_setup()
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 341, in run_setup
          exec(code, locals())
        File "<string>", line 1, in <module>
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/__init__.py", line 107, in setup
          return distutils.core.setup(**attrs)
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 159, in setup
          dist.parse_config_files()
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 895, in parse_config_files
          setupcfg.parse_configuration(
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/config/setupcfg.py", line 189, in parse_configuration
          meta.parse()
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/config/setupcfg.py", line 500, in parse
          section_parser_method(section_options)
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/config/setupcfg.py", line 475, in parse_section
          self[name] = value
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/config/setupcfg.py", line 293, in __setitem__
          parsed = self.parsers.get(option_name, lambda x: x)(value)
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/config/setupcfg.py", line 606, in _parse_version
          return expand.version(self._parse_attr(value, self.package_dir, self.root_dir))
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/config/setupcfg.py", line 417, in _parse_attr
          return expand.read_attr(attr_desc, package_dir, root_dir)
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/config/expand.py", line 196, in read_attr
          module = _load_spec(spec, module_name)
        File "/private/var/folders/82/phlx_f6j5518qtvpm4x0_0fm0000gq/T/pip-build-env-hwd34tc7/overlay/lib/python3.10/site-packages/setuptools/config/expand.py", line 216, in _load_spec
          spec.loader.exec_module(module)  # type: ignore
        File "<frozen importlib._bootstrap_external>", line 883, in exec_module
        File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
        File "/Users/genevieb/Documents/GitHub/micro-sam/./micro_sam/__init__.py", line 11, in <module>
          from .sample_data import (
        File "/Users/genevieb/Documents/GitHub/micro-sam/./micro_sam/sample_data.py", line 9, in <module>
          import imageio.v3 as imageio
      ModuleNotFoundError: No module named 'imageio'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```
<p>



</p>
</details>

Reply:
It usually works for me with
```
pip install -e .
```
but I haven't tried with the new `setup.cfg` yet. Will check it out and see if I can reproduce the problem.